mxfusion/inference/inference.py

```diff
@@ -58,6 +62,7 @@ def __init__(self, inference_algorithm, constants=None,
                                           dtype=self.dtype,
                                           context=self.mxnet_context)
         self._initialized = False
+        self._logger = logger
```

Comment:
On this line, can you create a default Logger object with no log_dir and verbose=True, if one isn't passed in?

Reply:
This will change the default behaviour for the `run()` method. Prior to this PR, this method would not print logs by default so are you saying you want to change this?

Reply:
Hmm no, thanks for noticing that.

So here's my conundrum. I don't want to change the interface such that people have to pass another object just to get printing. I would also like to keep the verbose argument in the run method.

Do you think it makes sense to change things such that:
1.  the `Logger` class doesn't have a `verbose` property, and just initializes as a passthrough (no MXBoard functionality) if it doesn't have `log_dir` passed. It would initialize without a `log_dir` by default, and if users want to use MXBoard they can construct a real logger and pass it.
2. the `log` method is passed a `verbose` argument instead. This means the `run` method in `Inference` keeps `verbose` as an argument as well and would just pass it down to `log`.

Reply:
Yes, I think that this is the best option given that you don't want users to have to pass an object to print logs